lib/src/OS/Windows/SharedMemory.cc

```diff
@@ -0,0 +1,69 @@
+#include <windows.h>
+#include <string>
+#include <stdexcept>
+
+#include "Error.h"
+#include "OS/SharedMemory.h"
+
+namespace kickcat
+{
+    #define  THROW_LAST_ERROR(msg) (throw std::system_error(static_cast<int>(GetLastError()), std::system_category(),  LOCATION(": " msg)))
+
+    SharedMemory::SharedMemory()
+        : fd_{nullptr}
+    {
+    }
+
+    SharedMemory::~SharedMemory()
+    {
+        if (address_ != nullptr)
+        {
+            if (not UnmapViewOfFile(address_))
+            {
+                THROW_LAST_ERROR("UnmapViewOfFile() failed");
+            }
+        }
+
+        if (fd_ != nullptr)
+        {
+            if (not CloseHandle(fd_))
+            {
+                THROW_LAST_ERROR("CloseHandle() failed");
+            }
+        }
+    }
+
+    void SharedMemory::open(std::string const& name, std::size_t size, void* address)
+    {
+        size_ = size;
+
+        // Create or open shared memory (file mapping in system paging file)
+        fd_ = CreateFileMappingA(
```

Comment:
I see a lot of people on stackoverflow using `CreateFileMappingW`, it seems to be the same thing. Here https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createfilemappinga vs https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-createfilemappingw
but i do not really know the subtleties. Just curious whats the difference ;)